definitions/checks/system_registration.rb

```diff
@@ -10,15 +10,23 @@ class Checks::SystemRegistration < ForemanMaintain::Check
   end
 
   def run
-    if system_is_self_registerd?
+    if warn?
       raise ForemanMaintain::Error::Warn, 'System is self registered'
     else
       puts 'System is not self registered'
     end
   end
 
-  def system_is_self_registerd?
-    rhsm_hostname.casecmp(hostname).zero?
+  def system_self_registerd?
+    @result ||= rhsm_hostname.casecmp(hostname).zero?
+  end
+
+  def warn?
+    if feature(:foreman_server).present?
+      !system_self_registerd?
```

Comment:
If I read it properly, we would warn if run on main server and the system was not self-registered. Despite the fact, that the error message would be `System is self registered`, is this what we actually want?